test/CompressedBlockPrefilteringTest.cpp

```diff
@@ -0,0 +1,547 @@
+
+  const std::vector<BlockMetadata> blocksWithDuplicate1 = {
+      b1,  b1,  b2,  b3,  b4,  b5,  b6,  b7,  b8,  b9,  b10, b11, b12,
+      b13, b14, b15, b16, b17, b18, b19, b20, b21, b22, b23, b24};
+
+  const std::vector<BlockMetadata> blocksWithDuplicate2 = {
+      b1,  b2,  b3,  b4,  b5,  b6,  b7,  b8,  b9,  b10, b11, b12, b13,
+      b14, b15, b16, b17, b18, b19, b20, b21, b22, b23, b24, b24};
+
+  // Function to create BlockMetadata
+  const BlockMetadata makeBlock(const ValueId& firstId, const ValueId& lastId) {
+    assert(firstId <= lastId);
+    static size_t blockIdx = 0;
+    ++blockIdx;
+    return {{{},
+             0,
+             // COLUMN 0  |  COLUMN 1  |  COLUMN 2
+             {firstId, VocabId10, DoubleId33, GraphId},  // firstTriple
+             {lastId, VocabId10, DoubleId33, GraphId},   // lastTriple
+             {},
+             false},
+            blockIdx};
+  }
+
+  // Check if expected error is thrown.
+  auto makeTestErrorCheck(std::unique_ptr<PrefilterExpression> expr,
+                          const std::vector<BlockMetadata>& input,
+                          const std::string& expected,
+                          size_t evaluationColumn = 0) {
+    AD_EXPECT_THROW_WITH_MESSAGE(expr->evaluate(input, evaluationColumn),
+                                 ::testing::HasSubstr(expected));
+  }
+
+  // Check that the provided expression prefilters the correct blocks.
+  auto makeTest(std::unique_ptr<PrefilterExpression> expr,
+                std::vector<BlockMetadata>&& expected) {
+    std::vector<BlockMetadata> expectedAdjusted;
+    // This is for convenience, we automatically insert all mixed blocks
+    // which must be always returned.
+    std::ranges::set_union(
+        expected, mixedBlocks, std::back_inserter(expectedAdjusted),
+        [](const BlockMetadata& b1, const BlockMetadata& b2) {
+          return b1.blockIndex_ < b2.blockIndex_;
+        });
+    ASSERT_EQ(expr->evaluate(blocks, 0), expectedAdjusted);
+  }
+};
+
+}  // namespace
+
+//______________________________________________________________________________
+TEST_F(TestPrefilterExprOnBlockMetadata, testBlockFormatForDebugging) {
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: I:0 V:10 D:33.000000 V:0\n(last) "
+      "Triple: I:0 V:10 D:33.000000 V:0\nnum. rows: 0.\n",
+      (std::stringstream() << b5).str());
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: I:-4 V:10 D:33.000000 V:0\n(last) "
+      "Triple: D:2.000000 V:10 D:33.000000 V:0\nnum. rows: 0.\n",
+      (std::stringstream() << b11).str());
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: V:14 V:10 D:33.000000 V:0\n(last) "
+      "Triple: V:17 V:10 D:33.000000 V:0\nnum. rows: 0.\n",
+      (std::stringstream() << b21).str());
+}
+
+// Test Relational Expressions
+//______________________________________________________________________________
+// Test LessThanExpression
+TEST_F(TestPrefilterExprOnBlockMetadata, testLessThanExpressions) {
```

Comment:
For clarity (althouh it is repeated a little) at the following at the beginning of every test:
```
// Note: the `makeTest` function automatically adds the blocks with mixed datatypes to the expected result.
```
(that's what I stumbled upon).